### DISCUSSION TITLE
Extend templates to lookup integrations, and lookup the integration that owns an entity

### DISCUSSION THREAD

**Discussion by erkr** [home-assistant/architecture#1092] [General]
***Intro:***
Recently templates have been extended with two useful functions to look up the entities owned by devices and entities owned by integrations. My proposal is to add two additional methods:
- get_integrations() -> list
- entity_integration(entity_id) -> string

***Reasoning:***
These methods would be useful to automate detection of devices and integrations that malfunction in a template (sensors or automations). When for example a device has multiple unavailable entities, one can easily check if those entities originate from the same integration or not.  
I also use this information to provide additional information in my dashboards for malfunctioning entities and devices:

<img src="https://github.com/home-assistant/architecture/assets/4363599/19c67a5f-395f-4936-9634-869a944ab2c8" height="400">

***My*** ***dirty*** ***workaround:***
Currently I retrieve the integration information via the *identifiers* attribute of the owning device. Not reliable as, not all entities belong to a device, and not all devices support the *identifiers* attribute. 
Alternatively I could look up entities of each integration. It's a little harsh to loop over integrations to check which integration a particular entity belongs to.

***Impact:***
I had a chat with @bdraco where he mentioned adding these two methods is just a matter of adding a few lines of code:
- `entity_integration(entity_id)` is effectively a wrapper around `entity_sources`  to expose a limited version .
- `get_integrations()` just returns `list(hass.config.all_components)`

Nick even demonstrated [a quick 'without tests' commit](https://github.com/home-assistant/core/commit/3270cadfab4286d5c830529ed204502895afdd40).

As I'm not experienced enough myself in python and GitHub delivery processes , but Nick offered to help out, in case this discussion results in acceptance. It's my first HA discussion, so be kind ;-)

Best Eric

**Comment by erkr** [comment]
More needed to get a conclusion on this proposal?

**Reply by Tahutipai** [reply]
+1 vote
